ui/src/components/NewWorkflow/Add.tsx

```diff
@@ -387,186 +322,62 @@ const Add: React.FC<AddProps> = ({
-                                      : 'newW.node.child',
-                                    intl
-                                  )} ${index + 1}`}
-                            </Typography>
-                          </Box>
-                        </Paper>
-                        {expand === index && (
-                          <Box mt={6}>
-                            <Add
-                              childIndex={index}
-                              parentTemplates={templates}
-                              setParentTemplates={setTemplates}
-                              setParentExpand={setExpand}
-                              externalTemplate={templates[index]}
-                            />
-                          </Box>
-                        )}
-                      </Box>
-                    ))}
+              {(values.type === 'serial' || values.type === 'parallel') && (
+                <>
+                  <SerialOrParallel
+                    name={values.name}
+                    deadline={values.deadline}
+                    type={values.type as TemplateType}
+                    childrenCount={values.num}
+                    submitTemplate={submit}
+                    templates={templates}
+                  ></SerialOrParallel>
+                </>
+              )}
+              {values.type === 'custom' && !isRenderedHTTPTask && (
+                <>
+                  <Task
+                    name={values.name}
+                    deadline={values.deadline}
+                    type={values.type as TemplateType}
+                    childrenCount={values.num}
+                    container={values.container}
+                    submitTemplate={submit}
+                    templates={templates}
+                    conditionalBranches={values.conditionalBranches}
+                  ></Task>
+                </>
+              )}
+              {isRenderedHTTPTask && (
+                <Box mt={3}>
+                  <HTTPTask
+                    name={values.name}
+                    type={values.type as TemplateType}
+                    childrenCount={values.num}
+                    templates={templates}
+                    submitTemplate={submit}
+                    externalTemplate={externalTemplate}
+                  />
                 </Box>
               )}
             </>
           )
         }}
       </Formik>
+
       {num < 0 && (
         <Box ml={8}>
-          <Box display={otherTypes ? 'none' : 'initial'}>
-            <NewExperimentNext ref={newERef} onSubmit={onSubmit} inWorkflow={true} />
-          </Box>
-          {otherTypes === 'suspend' && (
+          {typeOfTemplate === 'suspend' && (
             <Box mt={3}>
               <Suspend initialValues={initialValues} submit={submit} />
             </Box>
           )}
+
+          {typeOfTemplate === 'single' && (
+            <Box display={otherTypes === 'suspend' ? 'none' : 'initial'}>
```

Comment:
In this part, I used `otherTypes` because I thought that rendering a form with `typeOfTemplate === 'single'` might get lag (Because the `<NewExperiment />` component is a little bit complicated and re-rendering the component takes up the JS main thread). But now I find I may have paid too much attention to detail, at least so far I don't think I've noticed any significant lag during use. 🥲

So I think removing all `otherTypes` related code is safe. What do you think about it? @STRRL

Reply:
> So I think removing all `otherTypes` related code is safe. What do you think about it? @STRRL

OK! I will take a try.

Reply:
Done in 1e982bdd